frame/staking/fuzzer/src/submit_solution.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Fuzzing for staking pallet.
+//!
+//! # Running
+//!
+//! Run with `cargo hfuzz run submit_solution`. `honggfuzz`.
+//!
+//! # Debugging a panic
+//!
+//! Once a panic is found, it can be debugged with
+//! `cargo hfuzz run-debug submit_solution hfuzz_workspace/submit_solution/*.fuzz`.
+
+use honggfuzz::fuzz;
+
+use mock::Test;
+use pallet_staking::testing_utils::{
+	self, USER, get_seq_phragmen_solution, get_weak_solution, setup_chain_stakers,
+	set_validator_count, signed_account,
+};
+use frame_support::assert_ok;
+use sp_runtime::{traits::Dispatchable, DispatchError};
+
+mod mock;
+
+#[repr(u32)]
+#[allow(dead_code)]
+#[derive(Debug, Clone, Copy)]
+enum Mode {
+	/// Initial submission. This will be rather cheap
+	InitialSubmission,
+	/// A better submission that will replace the previous ones. This is the most expensive.
+	StrongerSubmission,
+	/// A weak submission that will be rejected. This will be rather cheap.
+	WeakerSubmission,
+}
+
+pub fn new_test_ext() -> Result<sp_io::TestExternalities, std::string::String> {
+	frame_system::GenesisConfig::default().build_storage::<mock::Test>().map(Into::into)
+}
+
+fn main() {
+	loop {
+		fuzz!(|do_reduce: bool| {
```

Comment:
I am getting some weird behaviours here -- running it just in a loop with `cargo run` works though. Should probably migrate to cargo fuzz since I am not even using the inputs.